scripts/azure-pipelines/test-modified-ports.ps1

```
@@ -142,16 +142,60 @@ $skipList = . "$PSScriptRoot/generate-skip-list.ps1" `
     -BaselineFile "$PSScriptRoot/../ci.baseline.txt" `
     -SkipFailures:$skipFailures
 
+$LogSkippedPorts = $true # Maybe parameter
+$changedPorts = @()
+$skippedPorts = @()
+if ($LogSkippedPorts) {
+    $diffFile = Join-Path $WorkingRoot 'changed-ports.diff'
+    Start-Process -FilePath 'git' -ArgumentList 'diff --name-only HEAD~10 -- versions ports' `
+        -NoNewWindow -Wait `
+        -RedirectStandardOutput $diffFile
+    $changedPorts = (Get-Content -Path $diffFile) `
+        -match '^ports/|^versions/.-/' `
+        -replace '^(ports/|versions/.-/)([^/]*)(/.*|[.]json$)','$2' `
+        | Sort-Object -Unique
+    $skippedPorts = $skipList -Split ','
+    $changedPorts | ForEach-Object {
+        if ($skippedPorts -contains $_) {
+            $port = $_
+            Write-Host "##vso[task.logissue type=error]$port`: build skipped, reason: CI baseline file."
+        }
+    }
+}
+
+$hostArgs = @()
 if ($Triplet -in @('x64-windows', 'x64-osx', 'x64-linux'))
 {
     # WORKAROUND: These triplets are native-targetting which triggers an issue in how vcpkg handles the skip list.
     # The workaround is to pass the skip list as host-excludes as well.
-    & "./vcpkg$executableExtension" ci $Triplet --x-xunit=$xmlFile --exclude=$skipList --host-exclude=$skipList --failure-logs=$failureLogs @commonArgs
-}
-else
-{
-    & "./vcpkg$executableExtension" ci $Triplet --x-xunit=$xmlFile --exclude=$skipList --failure-logs=$failureLogs @commonArgs
+    $hostArgs = @("--host-exclude=$skipList")
 }
+
+$current_port_and_features = ':'
+& "./vcpkg$executableExtension" ci $Triplet --x-xunit=$xmlFile --exclude=$skipList --failure-logs=$failureLogs @hostArgs @commonArgs `
+    | ForEach-Object {
+        $_
+        if ($LogSkippedPorts) {
+            if ($_ -match '^ *([^ :]+):[^:]*: *cascade:' -and $changedPorts -contains $Matches[1]) {

```

I'm a little concerned about making this "output intended for human consumption" load bearing in this way. Not an over my dead body issue but a better solution would be to have the tool do this processing and report it in the output so that changes in the tool don't cause silent failure of the infrastructure.

That said, while that is a solution I prefer, this solution is better than the status quo and such failure would only degrade things to the status quo, so I would not block merging over that.

---

```suggestion
            # Parse the ports ABI hash list at the start of output looking for ports vcpkg ci wants to build
            # (denoted by *) which are marked "cascade" because one of their dependencies were in
            # ci.baseline.txt.
            if ($_ -match '^ *([^ :]+):[^:]*: *cascade:' -and $changedPorts -contains $Matches[1]) {

```